frontend/src/views/admin/ddah-table/index.tsx

```diff
@@ -54,6 +55,33 @@ export function getReadableStatus(ddah: Pick<Ddah, "status">) {
     }
 }
 
+function isIsoDateString(dateString: string | null) {
+    if (dateString == null) return false;
+    if (!/\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z/.test(dateString))
+        return false;
+    const date = new Date(dateString);
+    return date.toISOString() === dateString;
+}
+
+function getRecentActivityDate(ddah: Ddah) {
+    const validDates = [
+        ddah.accepted_date,
+        ddah.approved_date,
+        ddah.emailed_date,
+        ddah.revised_date,
+    ]
+        .filter((dateString) => {
+            return isIsoDateString(dateString);
+        })
+        // dateString || 0 is required to avoid
+        // typescript complaining about potentially passing null to Date constructor,
+        // which cannot happen (although it complains) due to previous filtering
+        .map((dateString) => new Date(dateString || 0).getTime());
+    if (validDates.length === 0) return "";
+    const recentActivityDate = new Date(Math.max.apply(null, validDates));
```

Comment:
How about `new Date(Math.max(...[<all the dates>].filter(date => date).map(date => new Date(date))))`

Reply:
But Typescript will complain that `Argument of type 'Date' is not assignable to parameter of type 'number'` when using `Math.max`, which is why I convert `Date`s to numeric values using `getTime`

Reply:
In that case, you can cast them all to numbers (use `+` as a prefix to cast), or you can retype `Max`. E.g.,

```
let x = new Date();
let y = new Date();

let z =(Math.max as (...values: any[]) => number)(x,y);
```